### PR TITLE
Integrate new params, experimental design (sample sheet), irt generation and library merging

### PR DESCRIPTION
**Pull request by Leon-Bichmann**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated
- [x] If necessary, also make a PR on the [nf-core/diaproteomics branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/diaproteomics)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
can you post the help text here to easier check how the sheet format is explained?

**Comment by Leon-Bichmann** [comment]
Screen Shot 2020-09-17 at 12 58 48

**Comment by Leon-Bichmann** [comment]
Screen Shot 2020-09-17 at 13 00 31

**Comment by Leon-Bichmann** [comment]
Screen Shot 2020-09-17 at 13 01 10

**Comment by Leon-Bichmann** [comment]
ok so, my idea so far was to let the user specify either those three inputs as sample sheets.

The files are grouped and processed together by the sample name meta-info column.

Or otherwise - if there is no library and instead it should be generated from DDA runs then there is the following other option:

**Comment by Leon-Bichmann** [comment]
Screen Shot 2020-09-17 at 13 03 26

**Comment by timosachsenberg** [comment]
> ok so, my idea so far was to let the user specify either those three inputs as sample sheets. Or otherwise - if there is no library and instead it should be generated from DDA runs then there is the following other option:

Don't exactly understand the logic. Is the generation from DDA runs just an additional step before running the other steps? Or is this an exclusive branch of the workflow the just produces the spectral library? I think this is not clear.

**Comment by Leon-Bichmann** [comment]
> > ok so, my idea so far was to let the user specify either those three inputs as sample sheets. Or otherwise - if there is no library and instead it should be generated from DDA runs then there is the following other option:
> 
> Don't exactly understand the logic. Is the generation from DDA runs just an additional step before running the other steps? Or is this an exclusive branch of the workflow the just produces the spectral library? I think this is not clear.

Ok, so for the OpenSwathWorkflow DIA search you can either use your own library generated previously or you might have DDA identifications that you want to turn into a library to search corresponding DIA data ("pseudo targeted").

Now the workflow provides both options :
if you set ```--generate_spectral_library false ```, you have to specify your own library using ```--input_spectral_library```
if you set ```--generate_spectral_library true ```, you have to specify dda runs and ids using ```--input_sheet_dda```

I hope this is clearer now, otherwise we can briefly discuss in the SIG meeting.

**Comment by Leon-Bichmann** [comment]
This workflow sketch might also be usefull:

Screen Shot 2020-09-17 at 13 29 22

**Comment by timosachsenberg** [comment]
I think I understand but I guess the documentation needs to be extended. The sketch is helpful

**Comment by Leon-Bichmann** [comment]
@timosachsenberg, I added the naming scheme from the OpenMS Experimental Design. I will merge the sample sheets (as discussed) at a later stage.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,16 @@
...
 
+## v1.0dev - [15.09.20]
+
+### `Added`
+
+Integrated sample sheet eg. experimental design
+Sample sheets can be supplied for dia and dda raw data, id and irt libraries
+more parameters
+RawFile parsing
+Pseudo iRT generation
```
Comment (ggabernet) [inline CHANGELOG.md:14]:
```suggestion
- [#34](https://github.com/nf-core/diaproteomics/pull/34) Integrated sample sheet eg. experimental design
- [#34](https://github.com/nf-core/diaproteomics/pull/34) Sample sheets can be supplied for dia and dda raw data, id and irt libraries
- [#34](https://github.com/nf-core/diaproteomics/pull/34) Added parameters: merge_libraries, min_overlap_for_merging, generate_pseudo_irts, n_irts, irt_n_bins, irt_min_bins_covered
- [#34](https://github.com/nf-core/diaproteomics/pull/34) RawFile parsing
- [#34](https://github.com/nf-core/diaproteomics/pull/34) Pseudo iRT generation
```

File: .github/workflows/awsfulltest.yml [hunk 2]
```diff
@@ -3,6 +3,9 @@ name: nf-core AWS full size tests
 # It runs the -profile 'test_full' on AWS batch
 
 on:
+  push:
+    branches:
+      - master
```
Comment (ggabernet) [inline .github/workflows/awsfulltest.yml:8]:
```suggestion
```

File: .github/workflows/awstest.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ on:
   push:
     branches:
       - master
+      - dev
```
Comment (ggabernet) [inline .github/workflows/awstest.yml:9]:
```suggestion
```

**Review by ggabernet — COMMENTED** [review]
It looks good to me from the content, I just got some suggestions. If you wanna test the AWS tests maybe let's make a small separate PR for that, so that all tests pass for this one

**Review by jpfeuffer — APPROVED** [review]
The overall code looks good. The workflow also looks meaningful but I am certainly not a DIA expert.
Some minor things:
- I think you should use `path` instead of `file` for the input/output channels for processes. It is safer once you have URLs in your sample sheets.
- You could think about using a hash of a row or a part of a row (e.g. the filename) to link different parts of the sample sheet (or if you decide to merge the sample sheets, which I think is a good idea, it probably won't matter)
- It should be tested on a "real size" dataset, you can use our ftp if the files are too big for github. Just put the ftp URLs into your sample sheet.

**Comment by Leon-Bichmann** [comment]
ok so I will merge this into to dev to incorporate the current changes.

I'll include the file to path changes in another PR, in principle the test profile is set for a full size test now.

**Pull request merged**